soccer/src/soccer/strategy/agent/position/pivot_test.cpp

```diff
@@ -0,0 +1,76 @@
+#include "pivot_test.hpp"
+
+namespace strategy {
+
+Pivot::Pivot(int r_id) : Position{r_id, "Pivot"} {}
+
+std::optional<RobotIntent> Pivot::derived_get_task(RobotIntent intent) {
+    // Get next state, and if different, reset clock
+    State new_state = next_state();
+    if (new_state != current_state_) {
+        curr_pt_ = last_world_state_->get_robot(true, robot_id_).pose.position();
+    }
+    current_state_ = new_state;
+
+    // Calculate task based on state
+    return state_to_task(intent);
+}
+
+std::string Pivot::get_current_state() {
+    return std::string{"Pivot"} + std::to_string(static_cast<int>(current_state_));
+}
+
+Pivot::State Pivot::next_state() {
+    // handle transitions between current state
+    switch (current_state_) {
+        case OUR_GOAL: {
```

Comment:
clang-tidy wants you to collapse these cases. but i'm okay if you don't. it's up to you